src/basic/strxcpyx.c

```diff
@@ -93,7 +93,7 @@ size_t strscpyl(char *dest, size_t size, const char *src, ...) {
         do {
                 size = strpcpy(&s, size, src);
                 src = va_arg(va, char *);
-        } while (src != NULL);
+        } while (src != NULL && size > 0);
```

Comment:
Even if strpcpy() handled size == 0, these functions should cease iterating when no size remains IMHO.